### DISCUSSION TITLE
Display a variable on a (Neo) segment display / Cobra Board

### DISCUSSION THREAD

**Discussion by worldpeace-germany** [missionpinball/missionpinball#126] [Q&A] [answered]
Hello,

I am following this https://missionpinball.org/hardware/opp/cobrapin/cobrapin_serial_segment_displays/ guide. Then using a segment_display_player like here https://missionpinball.org/config/segment_display_player/ to display something on the segment. I am using a Cobra board the segment displays from Cobra. The basics are working fine, based on an event I can display a hard coded text like "Hello" on the display.

Now I want to display the current ball, score, etc. on these displays. Now I have two problems:

a) I have an event and based on that event I want to display the variable (ball). I thought instead of the hardcoded text I can use (ball), but that doesn't work.

b) When the score gets increased I believe there is no event. So how can I display the score on these displays?

Thanks!

**Comment by BenTripoli** [comment, marked as answer]
There is more on this on the google mpf users forum. But this is my code below, show player is used instead of segment player as there is a bug in segment_player. 
Below mine will show ball number and 66 credits. It will update your scores etc. Most of the below was posted on the google forum so have a look there if this doesnt help. I basically copied the code and jigged it with my own options and it worked fine.
it seems the formatting is getting removed when I post the below code. If you need anything more or the file itself I can post that.

This is the code from Cobra 
https://github.com/cobra18t/CobraPin/blob/main/MPF%20Config/ExampleGames/TagTeam_example.zip

mode:
  start_events: ball_starting #
  stop_events: game_ended
  stop_on_ball_end: false
  game_mode: false
  priority: 300

show_player:
  ball_started:
    ballnum:
      show_tokens:
        txt: "{(current_player.ball):d}  66"
  ball_started{player==1 and ball==1}:
    p1_score:
      show_tokens:
        txt: "{(players[0].score):d}"
  player_score{player_num==1}:
    p1_score:
      show_tokens:
        txt: "{(players[0].score):d}"
  player_score{player_num==2}:
    p2_score:
      show_tokens:
        txt: "{(players[1].score):d}"
  player_score{player_num==3}:
    p3_score:
      show_tokens:
        txt: "{(players[2].score):d}"
  player_score{player_num==4}:
    p4_score:
      show_tokens:
        txt: "{(players[3].score):d}"


  player_added{num==2}:
    p2_score:
      show_tokens:
        txt: "      0"
  player_added{num==3}:
    p3_score:
      show_tokens:
        txt: "      0"
  player_added{num==4}:
    p4_score:
      show_tokens:
        txt: "      0"

shows:
  ballnum:
    - duration: -1
      segment_displays:
        neoSeg4:   # ball number and credit seg display
          text: (txt)
  p1_score:
    - duration: -1
      segment_displays:
        neoSeg0:
          text: (txt)
  p2_score:
    - duration: -1
      segment_displays:
        neoSeg1:
          text: (txt)
  p3_score:
    - duration: -1
      segment_displays:
        neoSeg2:
          text: (txt)
  p4_score:
    - duration: -1
      segment_displays:
        neoSeg3:
          text: (txt)

**Comment by worldpeace-germany** [comment]
Thanks, that solved my problem. My main problem was that the event player_score you are using is not in the Event Reference https://missionpinball.org/events/ I would have expected in there. Anyways, problem solved. I will update the documentation accordingly.
